maitake/src/wait/cell.rs

```diff
@@ -7,14 +7,35 @@ use crate::{
         },
     },
     util::tracing,
-    wait::{self, WaitResult},
 };
 use core::{
+    future::Future,
     ops,
-    task::{Poll, Waker},
+    pin::Pin,
+    task::{Context, Poll, Waker},
 };
 use mycelium_util::{fmt, sync::CachePadded};
 
+/// An error indicating that a [`WaitCell`] or [`WaitQueue`] was closed while
+/// attempting register a waiter.
```

Comment:
i think this doc comment needs to be updated --- this is only returned by `WaitCell`, and can indicate that the cell is closed _or_ busy.